### ISSUE TITLE
Payment notifications

### ISSUE THREAD

**Issue by rkfg** [hieblmi/go-host-lnaddr#4]
It's currently hard to find if you've been paid. The only way is listing all paid invoices and look for those you don't remember creating yourself. A simple abstract notification interface with few implementations would be great to have (I might implement it myself if I find time). For example, it could be e-mail notification and http hook. I like how [lightningtip does it](https://github.com/michael1011/lightningtip/blob/master/notifications/mail.go). Shouldn't be hard to do.

**Comment by hieblmi** [comment]
Hi rkfg,

Thank you for your continued interest!

Yes I agree, we should have a better way of knowing when a payment was
received.

I recently released a go-module that allows you to listen to routing events
through your lnd node. I use this mod in a telegram notifier to get push
messages about routing events.

https://github.com/hieblmi/go-lnd-router-events (observable module to
listen to forwarding events)
https://github.com/hieblmi/telegram-lnd-events (Telegram Bot that uses this
module and sends it to your bot)

We could modify the use go-lnd-router-events and integrate it with the
hosting of the lightning address. And then publish it anywhere we want.
There is a service that allows you to generate generic endpoints on the
web... https://beeceptor.com/console/nicehash
We could test it with that.

Let me know what you think.

Best,
Michael

P.S.: Trying to pick up more work here but it's hard next to a full time
job and other commitments.

**Comment by rkfg** [comment]
I think that event observer won't work here. For some reason lnd gRPC protocol doesn't have the amount field in the incoming payment events. That field is only present in forwarding and outgoing payment events which isn't the case here. Lightningtip listens for invoice settlements, however, and getting the amount from there should be easier. Also need to filter those invoices by hash so we don't accidentally spam the user with unrelated settlements. [This method](https://github.com/lightningnetwork/lnd/blob/fbb1d159e089f8ce3beaf351efe06f78b0dcdc1a/lnrpc/lightning.proto#L318) is used for this purpose.

Regarding Telegram, I don't use it and will never recommend it to anyone, especially Bitcoiners. It's a very shady FOSS-washing (FOSS on paper but not in reality, just code dumps with zero community involvement, actually hostile towards community and FOSS ideals) 100% centralized (no federation support) messenger that tightly cooperates with Russian special services, poor privacy (requires phone number to register, phones were leaked multiple times), security (in-house buggy encryption schemes) and unknown funding sources. If you want to have good privacy (more or less because it's always a compromise between that and usability) I'd suggest installing your own Matrix server and using Element. A Matrix notification bot would be another good implementation and the SDK is pretty trivial.

**Comment by hieblmi** [comment]
I agree on the take on Telegram, was just playing around with it for the first time.

And yes - it looks like upon payment receipt only a lnd SettleEvent(stating our preimage) is triggered but not ForwardEvents that we could read the amounts etc. from.
The method for related invoice filtering you describe makes sense to me. For my purposes I'd like to integrate the invoice settlement stuff into https://github.com/hieblmi/go-lnd-router-events. Clients could then additionally subscribe to invoice settlements. Will look into your reference of Lightningtip when I get some time :-).
